### ISSUE TITLE
readiness probe for play kube

### ISSUE THREAD

**Issue by pkliczewski** [containers/podman#12417]
/kind feature

In pod spec as defined by Kuberenetes there is [readiness](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-readiness-probes) probe. ATM the support is missing in podman play kube.

**Comment by flouthoc** [comment]
@pkliczewski I worked on liveness probe a while ago since we dont have `kubelet` here it was backed by native podman health checks / systemd and restarts container if they need to. https://github.com/containers/podman/pull/10956

I might be missing on something but i am still curious what practical application would `readiness` probe would solve here since we don't have a construct of `service` in podman and there is no loadbalancer which will pass traffic to only `ready` pods.

**Comment by pkliczewski** [comment]
When multiple containers are booting and some depend on others with readiness probe it would be easy to understand whether application would work as a whole.

**Comment by flouthoc** [comment]
@pkliczewski SGTM,  I  think we would need another pod **field** in podman. As right now it shows if a `pod` is `Running, Created, Exited, Stopped, Dead` there is no **Ready** field like `kubernetes`.  

>When multiple containers are booting and some depend on others with readiness probe it would be easy to understand whether application would work as a whole.

This is a fair request I agree if `n/n` containers are ready in theory application should work.

**Comment by mheon** [comment]
This doesn't sound like a pod state to me - possibly a bool in `podman pod inspect` that states if the probes are healthy (`Ready: true` or `Ready: false`).

**Comment by flouthoc** [comment]
> need another pod field in podman.

@mheon Yes I intended the same a new field other than `pod state`. I guess it was not clear enough.

**Comment by rhatdan** [comment]
@umohnani8 FYI

**Comment by rhatdan** [comment]
If Podman play kube launched a pod that required another pod to be ready, would it wait for the readiness probe to complete?

**Comment by pkliczewski** [comment]
> If Podman play kube launched a pod that required another pod to be ready, would it wait for the readiness probe to complete?

It depends whether the readiness probe is used by podman or by a pod waiting on required dependency. I think it should be the latter.

**Comment by rhatdan** [comment]
@umohnani8 PTAL

**Comment by rhatdan** [comment]
@umohnani8 Any progress?

**Comment by baude** [comment]
Im tracking this for a 4.1 feature; will make a jira card.

**Comment by rhatdan** [comment]
@baude @umohnani8 Progress?

**Comment by baude** [comment]
it was just assigned last week.  and the function has to first go into podman/libpod.

**Comment by umohnani8** [comment]
@mheon this is what you have been working on with https://issues.redhat.com/browse/RUN-1432 right?

**Comment by mheon** [comment]
I'll have to verify

**Comment by DomenPigeon** [comment]
Just curious if there was any progress on this, I see that the https://issues.redhat.com/browse/RUN-1432 issue was closed.
